api/app/src/routes/fhir-proxy.ts

```diff
@@ -1,16 +1,68 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import proxy from "express-http-proxy";
 
+const URN_OID_PREFIX = "urn:oid:";
+
+const updateDocumentReferenceQueryString = (params: string): string => {
+  const decodedParams = decodeURIComponent(decodeURI(params));
+  return (
+    decodedParams
+      .replace(/patient\.identifier/i, "patient")
+      // eslint-disable-next-line no-useless-escape
+      .replace(/urn\:oid\:.+\|(2\.\d+)/g, "$1") // remove orgId from doc query's patient id
+  );
+};
+const updateQueryString = (path: string, params: string): string | undefined => {
+  if (path.toLocaleLowerCase().includes("documentreference")) {
+    return updateDocumentReferenceQueryString(params);
+  }
+  return undefined;
+};
+
 // TODO make this dynamic/config/secret
-const router = proxy("https://fhir.staging.metriport.com", {
+// const router = proxy("https://fhir.staging.metriport.com", {
+const router = proxy("http://host.docker.internal:8888", {
   proxyReqPathResolver: function (req) {
-    return new Promise(function (resolve) {
-      const parts = req.url.split("?");
-      const updatedPath = parts[0];
-      const queryString =
-        parts.length > 1 ? parts[1].replace(/patient\.identifier/, "patient") : undefined;
-      const resolvedPathValue = updatedPath + (queryString ? "?" + queryString : "");
-      resolve("/fhir" + resolvedPathValue);
-    });
+    console.log(`ORIGINAL URL: `, req.url);
+    const parts = req.url.split("?");
+    const path = parts[0];
+    const queryString = parts.length ? updateQueryString(path, parts[1]) : undefined;
+    const updatedURL = "/fhir" + path + (queryString ? "?" + queryString : "");
+    console.log(`UPDATED URL: `, updatedURL);
+    return updatedURL;
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  userResDecorator: function (proxyRes, proxyResData, userReq, userRes) {
+    const payloadString = proxyResData.toString("utf8");
+    console.log(`ORIGINAL RESPONSE: `, JSON.stringify(JSON.parse(payloadString), undefined, 2));
+    // const orgRegex = /"Organization\/(.+)"/g;
+    // const orgReplace = `"Organization/urn:oid:$1"`;
+    // eslint-disable-next-line no-useless-escape
```

Comment:
Leaving some commented-out code while in draft, might need to re-enable these as we go about getting CW to process our responses.